classes/controller/ModuleAdminController.php

```diff
@@ -103,22 +103,8 @@ protected function getTemplateLookupPaths()
      */
     protected function l($string, $class = null, $addslashes = false, $htmlentities = true)
     {
-        $translated = parent::l($string, $class, $addslashes, $htmlentities);
+        @trigger_error(__FUNCTION__ . 'is deprecated. Use ModuleAdminController::trans instead.', E_USER_DEPRECATED);
 
-        if ($translated === $string) {
-            // legacy fallback
-
-            if ($class === null || $class == 'AdminTab') {
-                $class = get_class($this);
-            }
-
-            $translated = Translate::getModuleTranslation($this->module->name, $string, $class, null, $addslashes);
-        }
-
-        if ($htmlentities) {
-            $translated = htmlspecialchars($translated, ENT_QUOTES, 'utf-8');
-        }
-
-        return $translated;
+        return $this->trans($string);
```

Comment:
This won't work with legacy dictionaries. You need to go through `Translate::getModuleTranslation()` because legacy dictionaries require the module name and the class name where the method is called.

I'd rather investigate about where the double escape is being performed

Reply:
In fact `Translate::getModuleTranslation` didn't translate well then I'll search there why it doesn't work.

Reply:
@MeKeyCool FYI, before your change, every translation was working as expected, and the only issue was the double escape.